docs/config-lifecycle.md

```diff
@@ -0,0 +1,35 @@
+# Build Config Lifecycle
+Configuration in CircleCI is sent to the platform as a valid YAML string that should both adhere to the schema and be semantically sound. Checks on the shape of your config will be conducted prior to processing your build. Once your build configuration hits our system, it will follow the path detailed below.
+
+## Lifecycle of configuration in a build
+
+Once a project is created with configuration, each build is run from that source of configuration in the following ways:
+
+1. When a build is initiated via webhook from your project's main git repository or via the API, the first check is of authorization.
+The credentials associated with the build request must be able to execute against the project. If credentials do not pass the authorization screen, you will not receive any confirmation that the project is even valid. This is similar behavior to, for instance, the way GitHub serves a 404 in response to a request for a repository that you don't have access to, rather than acknowledge that the repository exists at all.
+
+2. If the build request passes authorization, the build increments the build number on the project and project configuration is retrieved from source based on the SHA1 of the commit (unless config is being passed in via the API).
+If your configuration cannot be retrieved the build will be failed and you will see a failed Build.
+
+3. Configuration for the build is validated to ensure it parses as YAML and adheres to the CircleCI [Configuration Schema](https://github.com/circleci/config-compilation/blob/master/resources/main/orbs.schema.json). 
```

Comment:
This seems like the wrong link. What is the right link?

Reply:
The schema is not yet public, so this still links to a private document.